packages/vda-node-manager/src/blockchain/userApi.ts

```diff
@@ -0,0 +1,1171 @@
+import { explodeDID } from '@verida/helpers';
+import EncryptionUtils from "@verida/encryption-utils";
+
+export interface IStorageNode {
+    name: string;
+    didAddress: string;
+    endpointUri: string;
+    countryCode: string;
+    regionCode: string;
+    datacenterId: BigNumberish;
+    lat: number;
+    long: number;
+    slotCount: BigNumberish,
+    establishmentDate: string;
+    acceptFallbackSlots: boolean;
+    status: EnumStatus;
+    unregisterTime?: string;
+    fallbackNodeAddress?: string;
+}
+
+export interface IFallbackNodeInfo {
+    fallbackNodeAddress: string;
+    availableSlots: BigNumberish;
+    fallbackProofTime: BigNumberish;
+    availableSlotsProof: BytesLike;
+}
+
+
+export class VeridaNodeManager {
+
+    protected config: VdaClientConfig
+    protected network: string
+    protected didAddress?: string
+
+    protected vdaWeb3Client? : VeridaContract
+
+    protected readOnly: boolean
+    protected contract?: ethers.Contract
+
+    protected CONTRACT_DECIMAL?: number
+    
+    public constructor(config: VdaClientConfig) {
+        if (!config.callType) {
+            config.callType = 'web3'
+        }
+
+        this.config = config
+        this.readOnly = true
+        if (!config.web3Options) {
+            config.web3Options = {}
+        }
+
+        this.network = config.network
+
+        if (config.callType == 'web3' && !(<Web3SelfTransactionConfig>config.web3Options).rpcUrl) {
+            (<Web3SelfTransactionConfig> config.web3Options).rpcUrl = <string> RPC_URLS[this.network]
+        }
+
+        const contractInfo = getContractInfoForNetwork("StorageNodeRegistry", this.network)
+
+        if (config.did) {
+            this.readOnly = false
+            const { address } = explodeDID(config.did)
+            this.didAddress = address.toLowerCase()
+
+            this.vdaWeb3Client = getVeridaContract(
+                config.callType, 
+                {...contractInfo,
+                ...config.web3Options})
+        } else {
+            let rpcUrl = (<Web3SelfTransactionConfig>config.web3Options).rpcUrl
+            if (!rpcUrl) {
+                rpcUrl = <string> RPC_URLS[this.network]
+            }
+
+            const provider = new JsonRpcProvider(rpcUrl)
+
+            this.contract = new Contract(contractInfo.address, contractInfo.abi.abi, provider)
+        }
+    }
+
+    /**
+     * Get a nonce from contract
+     * @returns nonce of DID
+     */
+    public async nonceFN() {
+        if (!this.vdaWeb3Client) {
+            throw new Error(`Config must specify 'did' or 'signKey'`)
+        }
+
+        const response = await this.vdaWeb3Client.nonce(this.didAddress);
+        if (response.data === undefined) {
+            throw new Error('Error in getting nonce');
+        }
+        return response.data;
+    }
+
+    /**
+     * Get DECIMAL of contract
+     * @returns DECIMAL of contract
+     */
+    public async getContractDecimal() {
```

Comment:
A couple of thoughts on the `getXX()` methods:

1. We could remove these and instruct developers to use `vda-node-client` for `getXX()` operations
2. These methods could remain here, but simply use `vda-node-client` methods for simplifying the code

Reply:
Pls chekc the `vda-node-manager` in `read-only` mode.

Reply:
Please clarify, I don't understand how your comment helps address my comments.

Reply:
I recommend the 2nd way.
While using the `vda-node-manager` instance, developers can call the `getXX` functions. It might be inconveninent if they create another instance to call `getXX` functions.